File: src/java/SelectionSort.java
import java.util.Arrays;
import java.util.Scanner; 
import java.lang.*;

class SelectionSort {
	public static void main(String[] args) {
		Scanner entrada = new Scanner(System.in);
		String[] numeros = entrada.nextLine().split(" ");

		int[] array = new int[numeros.length];

		for (int i = 0; i < array.length; i++) {
			array[i] = Integer.parseInt(numeros[i]);
		}

		selectionSort(array);

	}

	public static void selectionSort(int[] array) {
		for (int i = 0; i < array.length; i++) {
			int i_menor = i;
			for (int j = i_menor + 1; j < array.length; j++) {
				if(array[j] < array[i_menor]) { 
					i_menor = j;
				}
			}	
			int aux = array[i];
			array[i] = array[i_menor];
			array[i_menor] = aux;
			
			
			if(i_menor != i) {
				System.out.println(Arrays.toString(array));
			}
		}
	}
}

File: src/javascript/selectionSort.js

function selectionSort(array) {
    let size = array.length;
    for (let i = 0; i < size; i++) {
        let min = i;
        for (let j = i + 1; j < size; j++) {
            if (array[min] > array[j]) {
                min = j;
            }
        }
        if (min !== i) {
            let tmp = array[i];
            array[i] = array[min];
            array[min] = tmp;
        }
    }
    return array;
}

let array = [10,8,3,4,15,42]
selectionSort(array)
console.log(array)